core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -399,7 +399,8 @@ private[spark] object JsonProtocol {
         ("Full Stack Trace" -> exceptionFailure.fullStackTrace) ~
         ("Accumulator Updates" -> accumUpdates)
       case taskCommitDenied: TaskCommitDenied =>
-        ("Job ID" -> taskCommitDenied.jobID) ~
+        ("Job ID" -> taskCommitDenied.stageID) ~
+        ("Job Attempt Number" -> taskCommitDenied.stageAttempt) ~
```

Comment:
Why does this use "Job" and not "Stage"?

Reply:
Also, will this affect the compatibility of the history server files?

Reply:
For the new property, I'm just following what the old property says, even though it's wrong. I think having `Job ID` and `Stage Attempt Number` would just be even more confusing...

Reply:
And it shouldn't affect compatibility. Given the code, even an old history server would be able to read these new log files.